frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.jsx

```diff
@@ -474,18 +475,22 @@ const LoadingState = () => (
 );
 
 const NoMatchState = ({ fields }) => {
-  if (fields.length > 1) {
-    // if there is more than one field, don't name them
-    return <OptionsMessage message={t`No matching result`} />;
+  if (fields.length === 1) {
+    const [{ display_name }] = fields;
+
+    if (display_name.length <= FIELD_TITLE_MAX_LENGTH) {
+      return (
+        <OptionsMessage
+          message={jt`No matching ${(
+            <strong>&nbsp;{display_name}&nbsp;</strong>
+          )} found.`}
+        />
+      );
+    }
   }
-  const [{ display_name }] = fields;
-  return (
-    <OptionsMessage
-      message={jt`No matching ${(
-        <strong>&nbsp;{display_name}&nbsp;</strong>
-      )} found.`}
-    />
-  );
+
+  // if there is more than one field, don't name them
```

Comment:
This comment is not 100% accurate after refactoring - it also handles a case with 0 fields.